both/collections/scheduleEvent.js

```diff
@@ -0,0 +1,91 @@
+
+// Add functions to the Mongo object, using transform (see http://docs.meteor.com/#/full/mongo_collection)
+ScheduleEvent = function(doc) {
+  _.extend(this, doc);
+};
+
+ScheduleEvent.MIN_DURATION = moment.duration(30, 'minutes');
+ScheduleEvent.DEFAULT_DURATION = moment.duration(60, 'minutes');
+
+_.extend(ScheduleEvent.prototype, {
+
+  endMinutes: function() {
+    return this.startMinutes + this.durationMinutes;
+  },
+  isScheduled: function() {
+    // Events scheduled for times outside of the range shown by our calendar
+    // are considered unscheduled. // TODO Maybe this wasn't broken before but it is now.  - Lars
+    var competition = Competitions.findOne(this.competitionId, {
+      fields: {
+        numberOfDays: 1,
+        calendarStartMinutes: 1,
+        calendarEndMinutes: 1,
+      }
+    });
+    assert(competition);
+    if(this.nthDay < 0 || this.nthDay >= competition.numberOfDays) {
+      return false;
+    }
+    if(this.endMinutes() < competition.calendarStartMinutes) {
+      return false;
+    }
+    if(this.startMinutes > competition.calendarEndMinutes) {
+      return false;
+    }
+    return true;
+  },
+});
+
+ScheduleEvents = new Mongo.Collection("scheduleEvents", { transform: function(doc) { return new ScheduleEvent(doc); } });
+
+ScheduleEvents.attachSchema({
+  competitionId: {
+    type: String,
+  },
+
+  roundId: {
+    type: String,
+    optional: true,
+//    index: true, // Bug reported: https://github.com/aldeed/meteor-collection2/issues/212
```

Comment:
Thanks for reporting. I added a comment here: https://github.com/aldeed/meteor-collection2/issues/212#issuecomment-96121119.

If I'm understanding this correctly, we should be able to add a sparse, unique index on roundId by calling `_ensureIndex` ourselves (the following code is totally untested and copied from [here](http://stackoverflow.com/a/13391188)):

```javascript
  ScheduleEvents._ensureIndex({
    roundId: { unique: true, sparse: true }
  });
```